internal/genconfig/generate_config_test.go

```diff
@@ -670,20 +768,20 @@ resource "tfcoremock_simple_resource" "example" {
 		},
 	}
 	for name, tc := range tcs {
-		t.Run(name, func(t *testing.T) {
+		t.Run(name, func(tt *testing.T) {
```

Comment:
I expect we should probably discuss this in another place since it's not directly relevant to this PR, but I think this is one place where shadowing the parent scope's `t` is both idiomatic style _and_ wise, because it's never correct to access the parent `t` from inside a `t.Run`... if you're lucky it causes a panic, and if you're less lucky it makes the test harness get confused.

I accept that we've decided to differ from typical Go idiom in a number of different ways in this project, but this particular one seems like an unnecessary deviation from how tests are typically written in Go and seems more likely to _cause_ problems than to avoid them. :face_with_diagonal_mouth:

Reply:
I'm super happy to move back to shadowing :D You're right and honestly I really dont mind either way. I'll update this.